src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseReader.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Data.Common;
+
+namespace Microsoft.ML.Data
+{
+    public abstract class DatabaseReader : IDisposable
```

Comment:
I think we could iterate on the name here some. I don't think `DatabaseReader` is a good name, because this doesn't actually do the reading.

Maybe `DatabaseFactory` or similar?

Actually, as I type this, I now remember: [https://docs.microsoft.com/en-us/dotnet/api/system.data.common.dbproviderfactory.createconnection](https://docs.microsoft.com/en-us/dotnet/api/system.data.common.dbproviderfactory.createconnection).  What if we just took a `DbProviderFactory` instance instead of defining our own new type?

Reply:
I agree, I don't feel comfortable by creating our own new type for an abstraction or factory if there's something comparable available already in system.data such as DbProviderFactory.